src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,232 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2024 SECO Mind Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Card, Container, Table } from 'react-bootstrap';
+import {
+  AstarteDataTuple,
+  AstarteDatastreamIndividualData,
+  AstarteDatastreamObjectData,
+  AstarteInterface,
+  AstarteInterfaceValues,
+} from 'astarte-client';
+import _ from 'lodash';
+
+import BackButton from './ui/BackButton';
+import { useAstarte } from './AstarteManager';
+import { AlertsBanner, useAlerts } from 'AlertManager';
+import 'react-datepicker/dist/react-datepicker.css';
+import RenderAstarteData from 'components/RenderAstarteData';
+import useFetch from 'hooks/useFetch';
+import FilterData from 'components/FilterData';
+
+function formatAstarteData(data?: AstarteDataTuple): string {
+  if (data == null) {
+    return '';
+  }
+  if (_.isArray(data)) {
+    return JSON.stringify(data);
+  }
+  if (_.isBoolean(data)) {
+    return data ? 'true' : 'false';
+  }
+  if (_.isNumber(data)) {
+    return data.toString();
+  }
+  if (_.isNull(data)) {
+    return '';
+  }
+  return String(data);
+}
+
+interface IndividualDatastreamTableProps {
+  data: AstarteDatastreamIndividualData[];
+}
+
+const IndividualDatastreamTable = ({
+  data,
+}: IndividualDatastreamTableProps): React.ReactElement => {
+  if (data.length === 0) {
+    return <p>Maybe device published data but not in the selected timeframe.</p>;
+  }
+
+  return (
+    <>
+      <Table responsive>
+        <thead>
+          <tr>
+            <th>Path</th>
+            <th>Value</th>
+            <th>Timestamp</th>
+          </tr>
+        </thead>
+        <tbody>
+          {data.map((tree) => (
+            <tr key={tree.endpoint}>
+              <td>{tree.endpoint}</td>
+              <td>{formatAstarteData(tree)}</td>
+              <td>{new Date(tree.timestamp).toLocaleString()}</td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+    </>
+  );
+};
+
+interface ObjectDatastreamTableProps {
+  data: AstarteDatastreamObjectData[];
+}
+
+const ObjectDatastreamTable = ({ data }: ObjectDatastreamTableProps): React.ReactElement => {
+  if (data.length === 0) {
+    return <p>No data sent by the device.</p>;
+  }
+  const orderedData = _.orderBy(data, 'timestamp', 'desc');
```

Comment:
I would not order data, but rather display it in the same order returned by Astarte. This way it's easier to debug and inspect what's returned by Astarte.
We can force some order by clicking on the table header in a future PR